docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+
+* [Traditional Virtual Machines](https://en.wikipedia.org/wiki/Virtual_machine)
+* [Unikernels](http://unikernel.org/)
+* [Docker/OCI container](https://www.opencontainers.org)
+
+Edge Container is a novel concept that allows an effective packaging and lifecycle management of all these types of applications and it consists of two parts:
+
+* Edge Container Image (ECI) - an self-contained, binary representation of an Edge Container
+* Edge Container Object (ECO) - a live copy of an Edge Container
+
+ECOs are created by users binding a single ECI to a set of hardware and software resources (e.g. memory, CPU, I/O and networks) and we sometimes use the term ECO to refer to both -- an overall concept of an Edge Container (because it is an Edge COntainer) and a live copy. It will be fair to say that the primary purpose of EVE as an engine is to provide the best resource abstraction and an execution environment to a set of ECOs.
+
+ECI then is an application packaging and distribution mechanism that provides support for all 3 types of EVE applications while seamlessly integrating with Docker/OCI container specification and toolchains. The later part is especially important, since Docker tooling (e.g. Docker Desktop, Docker Registry, CI/CD integrations, etc.) has become a de-facto standard in modern, cloud-native application development. ECI format is a foundation for everything else in the Edge Containers workflow. This format defines the very notion of an ECO binary the same way that [ELF format](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format) defines what it means to be an application on most traditional Operating Systems today. And just like with ELF, it is the job of an operating system to load (L in ELF) and execute (E in ELF) these turning them into running ECOs.
+
+EVE TSC plans to submit ECO and ECI standards for broad Linux Foundation adoption once this document reaches the state of a formal specification. At the current stage, however, we are still collecting product requirements and recording them (together with implementation details) below.
+
+Working with ECOs will typically go through multiple stages, each with its own expectations around standardization and how they interact with the rest of the cloud-native developer and DevOps toolchains. We expect EVE users to...
+
+   #1 ...build an Edge Container Image (ECI) using the usual tools typically available on developer's desktop. The final build artifact may either be hidden in a local cache of some kind (along the lines of Docker image repository, local Maven cache, etc.) or be immediately available as a self-contained binary file
+   #2 ...instantiate ECOs based on a given ECI in a simulated environment that can easily run on developer's desktop (unlike a real EVE instance that typically runs on an Edge Node). Input to this stage could either be a reference to an ECI hidden in a local cache or a URL (including a local URL) for a self-contained binary file
```

Comment:
I like this part about simulation and contrasting to a real node. On the other hand, the second sentence detracts from the brief overview. Whether it is run from a local cache or some URL is immaterial to this ordered list.

Reply:
This is less of a brief overview and more of a brief overview/"product requirements"/vision doc at this point. I agree it makes it less clear in places -- but it is important to capture these types of details for now so we can refine it later.